specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json

```diff
@@ -0,0 +1,8511 @@
+        "resourceGroupName",
+        "resourceName",
+        "resourceLocation"
+      ],
+      "type": "object",
+      "properties": {
+        "subscriptionId": {
+          "description": "Arc resource subscription Id",
+          "type": "string"
+        },
+        "resourceGroupName": {
+          "description": "Arc resource group name",
+          "type": "string"
+        },
+        "resourceName": {
+          "description": "Arc resource Name",
+          "type": "string"
+        },
+        "resourceLocation": {
+          "description": "Arc resource location",
+          "type": "string"
+        },
+        "version": {
+          "description": "Arc resource version",
+          "type": "string",
+          "readOnly": true
+        },
+        "hostPlatform": {
+          "description": "Host OS supported by the Arc addon.",
+          "enum": [
+            "Windows",
+            "Linux"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "PlatformType",
+            "modelAsString": true
+          }
+        },
+        "hostPlatformType": {
+          "description": "Platform where the runtime is hosted.",
+          "enum": [
+            "KubernetesCluster",
+            "LinuxVM"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "HostPlatformType",
+            "modelAsString": true
+          }
+        },
+        "provisioningState": {
+          "description": "Addon Provisioning State",
+          "enum": [
+            "Invalid",
+            "Creating",
+            "Created",
+            "Updating",
+            "Reconfiguring",
+            "Failed",
+            "Deleting"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "AddonState",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "ARMBaseModel": {
+      "description": "Represents the base class for all object models.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "The path ID that uniquely identifies the object.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "The object name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "The hierarchical type of the object.",
+          "type": "string",
+          "readOnly": true
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "AsymmetricEncryptedSecret": {
+      "description": "Represent the secrets intended for encryption with asymmetric key pair.",
+      "required": [
+        "value",
+        "encryptionAlgorithm"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The value of the secret.",
+          "type": "string"
```

Comment:
This should be marked as x-ms-secret and nulled out when present in GET.

Reply:
Added.